### PR TITLE
Add `ignore_funding_requests` config flag

### LINKED ISSUE #5764: Funding message can not be disabled
### Describe the problem as clearly as you can

Bundler prints this after each run:
```
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
```

There seems to be no way to disable it.
In npm the message can be suppressed by
putting `fund=false` in `~/.npmrc`, bundler
should have an equivalent option.

### Did you try upgrading rubygems & bundler?

Yes.

```
Bundler version 2.3.18
```

### Post steps to reproduce the problem

1. Run `bundle`

### Which command did you run?

`bundle`

### What were you expecting to happen?

To not see any messages printed to my screen that are
unrelated to the operation I invoked.

### What actually happened?

I get extra lines in my local and CI output that I don't want to see.

### PR DESCRIPTION
**Pull request by m-o-e**
## What was the end-user or developer problem that led to this PR?

Bundler prints this after each run:
```
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
```

There seems to be no way to disable it (https://github.com/rubygems/rubygems/issues/5764).

## What is your fix for the problem, implemented in this PR?

Add `ignore_funding_requests` config flag.

Closes #5764.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [x] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

File: bundler/spec/install/gems/fund_spec.rb [hunk 1]
```diff
@@ -52,6 +52,35 @@
       end
     end
 
+    context "when gems include a fund URI but `ignore_funding_requests` is configured" do
+      let(:config) { "ignore_funding_requests true" }
+
+      before do
+        bundle "config set #{config}" if config
```
Comment (deivid-rodriguez) [inline bundler/spec/install/gems/fund_spec.rb:59]: Why the conditional? `config` should be always defined, right? Can you instead do `bundle "config set ignore_funding_requests true"` directly?

Comment (m-o-e) [reply bundler/spec/install/gems/fund_spec.rb]: Good point, I've changed it now.

I had just copied the first version from a different spec and
left it as is for consistency - but you are right, didn't make
sense for this one as the value is unlikely to ever change.

**Review by deivid-rodriguez — COMMENTED** [review]
This looks good to me, can you rebase and squash everything to a single commit?

**Comment by m-o-e** [comment]
> This looks good to me, can you rebase and squash everything to a single commit?

done :+1:

**Review by deivid-rodriguez — APPROVED** [review]
Looks good to me, including the name you chose. But I'll let this sit some days in case it gets some more feedback!

**Comment by m-o-e** [comment]
thanks 🙇

File: bundler/lib/bundler/man/bundle-config.1.ronn [hunk 2]
```diff
@@ -207,6 +207,8 @@ learn more about their operation in [bundle install(1)](bundle-install.1.html).
 * `ignore_messages` (`BUNDLE_IGNORE_MESSAGES`):
    When set, no post install messages will be printed. To silence a single gem,
    use dot notation like `ignore_messages.httparty true`.
+*  `ignore_funding_requests` (`BUNDLE_IGNORE_FUNDING_REQUESTS`):
```
Comment (simi) [inline bundler/lib/bundler/man/bundle-config.1.ronn:210]: Can we move it one entry up to keep it in alphabetical order?

Comment (m-o-e) [reply bundler/lib/bundler/man/bundle-config.1.ronn]: done :+1:

File: bundler/lib/bundler/man/bundle-config.1.ronn [hunk 3]
```diff
@@ -204,6 +204,8 @@ learn more about their operation in [bundle install(1)](bundle-install.1.html).
 * `global_gem_cache` (`BUNDLE_GLOBAL_GEM_CACHE`):
    Whether Bundler should cache all gems globally, rather than locally to the
    installing Ruby installation.
+*  `ignore_funding_requests` (`BUNDLE_IGNORE_FUNDING_REQUESTS`):
```
[the remark this answers is not in the document]
Comment (m-o-e) [reply bundler/lib/bundler/man/bundle-config.1.ronn]: Argh!  You are right. Sorry, no idea how that happened.
Force pushed again. 🙈

File: bundler/lib/bundler/man/bundle-viz.1 [hunk 4]
```diff
@@ -15,6 +15,9 @@
 .P
 The associated gems must also be installed via \fBbundle install(1)\fR \fIbundle\-install\.1\.html\fR\.
 .
+.P
+\fBviz\fR command was deprecated in Bundler 2\.2\. Use bundler\-graph plugin \fIhttps://github\.com/rubygems/bundler\-graph\fR instead\.
+.
```
Comment (deivid-rodriguez) [inline bundler/lib/bundler/man/bundle-viz.1:20]: This seems unrelated, problem with rebase maybe?

Comment (m-o-e) [reply bundler/lib/bundler/man/bundle-viz.1]: Not entirely sure what happened there either.
I think it may have to do with me having had merged master
into this branch earlier but then later rebasing it.

After another rebase it is gone now. Sorry for the noise 🤷

Comment (deivid-rodriguez) [reply bundler/lib/bundler/man/bundle-viz.1]: No worries, thanks!

**Comment by deivid-rodriguez** [comment]
@simi's feedback seems addressed, so merging! Thank you!
